diff --git a/configure.ac b/configure.ac
index 849eb2b..989283b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -215,9 +215,21 @@ AC_SUBST([AM_CFLAGS])
 AC_SUBST([AM_CXXFLAGS])
 AC_SUBST([AM_LDFLAGS])
 
-GTK_DOC_CHECK([1.14],[--flavour no-tmpl])
-GOBJECT_INTROSPECTION_CHECK([0.9.5])
-GNOME_CODE_COVERAGE
+AM_CONDITIONAL(ENABLE_GTK_DOC, test "x$enable_gtk_doc" = xyes)
+AM_CONDITIONAL(HAVE_INTROSPECTION, test "x$have_introspection" = xyes)
+m4_ifdef([GTK_DOC_CHECK], [
+ GTK_DOC_CHECK([1.14],[--flavour no-tmpl])
+], [
+ AC_SUBST(SCANOBJ_FILES, '')
+])
+m4_ifdef([GOBJECT_INTROSPECTION_CHECK], [
+ GOBJECT_INTROSPECTION_CHECK([0.9.5])
+])
+m4_ifdef([GNOME_CODE_COVERAGE], [
+ GNOME_CODE_COVERAGE
+], [
+ AC_SUBST(GNOME_CODE_COVERAGE_RULES, '')
+])
 
 AC_CONFIG_FILES([
 Makefile
